oneview/data_source_ethernet_network.go

```diff
@@ -124,7 +125,11 @@ func dataSourceEthernetNetworkRead(d *schema.ResourceData, meta interface{}) err
 	config := meta.(*Config)
 	name := d.Get("name").(string)
 	eNet, err := config.ovClient.GetEthernetNetworkByName(name)
-	if err != nil || eNet.URI.IsNil() {
+
+	if err != nil {
+		d.SetId("")
+		return err
+	} else if eNet.URI.IsNil() {
```

Comment:
don't we need to do null check for eNet itself before accessing its properties, I mean eNet != nil || eNet.URI.isNil, something like this